educator/src/Dscp/DB/SQLite/Functions.hs

```diff
@@ -362,13 +328,6 @@ sqlDebugLogger msg =
     , msg
     ]
 
--- | Enables SQLite tracing locally. For debug purposes.
---
--- Note: if any trace handler was set globally, it will be lost after that.
-traced :: MonadUnliftIO m => DBT t w m a -> DBT t w m a
-traced action = do
-    conn <- DBT ask
-    UIO.bracket_
-        (liftIO $ Backend.setTrace conn (Just $ sqlDebugLogger . toString))
-        (liftIO $ Backend.setTrace conn Nothing)
-        action
+-- | Enables SQL queries tracing locally. For debug purposes.
+traced :: MonadUnliftIO m => SqliteM a -> SqliteM a
+traced = SqliteM . local (_1 .~ sqlDebugLogger) . runSqliteM
```

Comment:
Please, write haskell, not APL. Also, is there really a _tuple_ there? Why not a proper type with a `smthLogger` lens?

Reply:
Because it is how `SqliteM` is defined (`ReaderT (Logger, Connection) Io`) :pepe: